source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -521,7 +524,68 @@ void RtCoreIo::read(const DocumentPtr& doc, RtCoreIo::ReadFilter filter)
     }
 }
 
-void RtCoreIo::write(DocumentPtr& doc, RtCoreIo::WriteFilter filter)
+void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPaths, RtFileIo::ReadFilter filter)
+{
+    try
+    {
+        DocumentPtr document = createDocument();
+        XmlReadOptions readOptions;
+        readOptions.skipConflictingElements = true;
+        readFromXmlFile(document, documentPath, searchPaths, &readOptions);
+
+        CopyOptions copyOptions;
+        copyOptions.skipConflictingElements = true;
+
+        read(document, filter);
+    }
+    catch(Exception&)
+    {
+        return;
+    }
+}
+
+void RtFileIo::loadLibraries(const StringVec& libraryPaths, const FileSearchPath& searchPaths, const RtToken& referenceName)
+{
+    // Always add in local search paths
+    FileSearchPath totalSearchPaths = searchPaths;
+    FilePath currentPath(FilePath::getCurrentPath());
+    FilePath parentCurrentPath = currentPath.getParentPath();
+    const FilePath librariesPath("libraries");
+    FilePath fullPath(currentPath / librariesPath);
+    if (!fullPath.exists())
+    {
+        fullPath = parentCurrentPath / librariesPath;
+        if (fullPath.exists())
+        {
+            totalSearchPaths.append(fullPath);
+        }
+    }
+    else
+    {
+        totalSearchPaths.append(fullPath);
+    }
+    totalSearchPaths.append(parentCurrentPath);
+    totalSearchPaths.prepend(currentPath);
```

Comment:
I'm not sure we should add the current path either, or any other path. Since the method takes a searchPath in a client will expect only the given search paths to be used.

Reply:
Agreed.